src/main/java/com/purbon/kafka/topology/KafkaTopologyBuilder.java

```diff
@@ -48,7 +46,11 @@ public void run() throws IOException {
     AccessControlProvider aclsProvider = buildAccessControlProvider();
     AccessControlManager accessControlManager = new AccessControlManager(aclsProvider);
 
-    TopicManager topicManager = new TopicManager(builderAdminClient);
+    SchemaRegistryClient schemaRegistryClient =
+        new CachedSchemaRegistryClient(properties.getProperty(SCHEMA_REGISTRY_URL), 10);
+    SchemaRegistryManager schemaRegistryManager = new SchemaRegistryManager(schemaRegistryClient);
+
+    TopicManager topicManager = new TopicManager(builderAdminClient, schemaRegistryManager);
```

Comment:
question out of the box: We have currently:

* A Topic Manager, that handles the actions regarding topics, I mean create, delete, update, configs, etc.
* An Access Control Manager, to do all access control things, aka ACLs and RBAC.

With your current proposal, schema registry is embed into topic management. What do you think about using the separate manager outside of the topic management?

I see for example one benefit, error handling and rollback, we could implement something like, catch exception from topic manager, and if then rollback anything coming out of SchemaRegistry, a kind of an internal saga. What do you think?

Reply:
They seem to be very tightly coupled. But I can have a look if you wish.